src/categories/categories.controller.ts

```diff
@@ -39,29 +47,76 @@ export class CategoriesController {
 
   @ApiResponse({
     status: HttpStatus.OK,
-    description: 'Get all categories',
+    description: 'Get categoriey by ID',
+    type: () => CategorySchema,
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: 'Category not found',
   })
   @Get(':id')
   getCategory(@Param('id') id: string) {
-    return this.categoriesService.getCategoryById(id);
+    const doc = this.categoriesService.getCategoryById(id);
+    if (!doc) {
+      throw new HttpException('No category found', HttpStatus.NOT_FOUND);
+    }
+    return doc;
   }
 
   @ApiResponse({
     status: HttpStatus.OK,
     description: 'Update category',
+    type: () => UpdateCategoryDto,
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: 'Category not found',
   })
   @Put(':id')
-  updateCategory(@Param('id') id: string, @Body() doc: CategoryDto) {
-    return this.categoriesService.updateCategory(id, doc);
+  async updateCategory(
+    @Param('id') id: string,
+    @Body() doc: UpdateCategoryDto,
+  ) {
+    try {
+      await this.categoriesService.updateCategory(id, doc);
+    } catch (e) {
+      if (e instanceof MongooseError) {
+        throw new HttpException('Category not found', HttpStatus.NOT_FOUND);
+      } else {
+        throw e;
+      }
+    }
   }
 
   @HttpCode(HttpStatus.NO_CONTENT)
   @ApiResponse({
     status: HttpStatus.NO_CONTENT,
-    description: 'Delete category',
+    description: 'Deleted category',
+  })
+  @ApiResponse({
+    status: HttpStatus.NOT_FOUND,
+    description: 'Category not found',
   })
   @Delete(':id')
-  deleteCategory(@Param('id') id: string) {
-    this.categoriesService.deleteCategory(id);
+  async deleteCategory(@Param('id') id: string) {
+    try {
+      await this.categoriesService.deleteCategory(id);
+    } catch (e) {
+      if (e instanceof MongooseError) {
+        throw new HttpException('Category not found', HttpStatus.NOT_FOUND);
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  @ApiResponse({
+    status: HttpStatus.OK,
+    description: 'Change category rank',
+    type: () => RankDto,
+  })
+  @Post('updateRank')
```

Comment:
Why updateRank is POST method

Reply:
because golf just send array to back we dont have anything to store this array in db it just a sequence of rank

Reply:
In category view. Frontend want to update the categories rank not create something in categories, so I think PATCH is more appropriate than POST